tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ProjectTemplateNameValidationService.java

```diff
@@ -14,56 +14,127 @@
 
 package com.liferay.ide.project.core.modules;
 
-import com.liferay.ide.core.util.ListUtil;
+import aQute.bnd.version.Version;
+import aQute.bnd.version.VersionRange;
 
-import java.util.Set;
+import com.liferay.ide.core.util.FileUtil;
+import com.liferay.ide.core.util.SapphireContentAccessor;
+import com.liferay.ide.project.core.ProjectCore;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.util.Enumeration;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+
+import org.apache.commons.io.IOUtils;
 
-import org.eclipse.sapphire.Event;
-import org.eclipse.sapphire.Listener;
-import org.eclipse.sapphire.Value;
 import org.eclipse.sapphire.modeling.Status;
 import org.eclipse.sapphire.services.ValidationService;
 
 /**
  * @author Gregory Amerson
+ * @author Terry Jia
  */
-public class ProjectTemplateNameValidationService extends ValidationService {
+public class ProjectTemplateNameValidationService extends ValidationService implements SapphireContentAccessor {
 
 	@Override
 	protected Status compute() {
-		return _templateNameStatus;
-	}
-
-	@Override
-	protected void initValidationService() {
-		super.initValidationService();
+		Status retval = Status.createOkStatus();
```

Comment:
Hey @simonjhy   Let's change this to not do as much work for every compute() call.

Instead during the `initValidationServce` you need to go through the blade.jar and get all of the project template versions and put them in some type of map or table.  And then during the `compute` you can get the value that the user has typed in and just access the map in memory instead of extracting all of the files and check again for a 2nd or 3rd time.